File: common/src/main/java/com/mcal/common/utilsOld/ApkInfoParser.java
package com.mcal.common.utilsOld;

import android.content.Context;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.graphics.drawable.Drawable;

import androidx.annotation.NonNull;

public class ApkInfoParser {

    public ApkInfoParser() {
    }

    public AppInfo parse(@NonNull Context ctx, String apkPath) throws Exception {
        AppInfo apkInfo = null;

        PackageManager packageManager = ctx.getPackageManager();
        PackageInfo packageInfo = packageManager.getPackageArchiveInfo(apkPath, 0);
        if (packageInfo != null) {
            apkInfo = new AppInfo();
            packageInfo.applicationInfo.sourceDir = apkPath;
            packageInfo.applicationInfo.publicSourceDir = apkPath;
            apkInfo.label = packageInfo.applicationInfo.loadLabel(packageManager).toString();
            apkInfo.packageName = packageInfo.packageName;
            apkInfo.icon = packageInfo.applicationInfo.loadIcon(packageManager);
        }

        return apkInfo;
    }

    public static class AppInfo {
        public String label;
        public String packageName;
        public Drawable icon;
    }
}

File: core/src/main/java/com/mcal/common/utils/ApkInfoParser.kt
package com.mcal.common.utils

import android.content.Context
import android.graphics.drawable.Drawable

class ApkInfoParser {
    @Throws(Exception::class)
    fun parse(ctx: Context, apkPath: String?): AppInfo? {
        var apkInfo: AppInfo? = null
        val packageManager = ctx.packageManager
        apkPath?.let { path ->
            packageManager.getPackageArchiveInfo(path, 0)?.let { packageInfo ->
                val appInfo = packageInfo.applicationInfo ?: return@let
                appInfo.sourceDir = path
                appInfo.publicSourceDir = path
                apkInfo = AppInfo().apply {
                    label = appInfo.loadLabel(packageManager).toString()
                    pkgName = packageInfo.packageName
                    icon = appInfo.loadIcon(packageManager)
                }
            }
        }
        return apkInfo
    }

    class AppInfo {
        @JvmField
        var label: String? = null

        @JvmField
        var pkgName: String? = null

        @JvmField
        var icon: Drawable? = null
    }
}